storage/s3/s3.go

```diff
@@ -165,18 +170,18 @@ func (s *SimpleStorageService) init(ctx context.Context) error {
 		case http.StatusForbidden:
 			err = fmt.Errorf(
 				"s3: insufficient permissions for accessing bucket '%s'",
-				s.bucket,
+				*s.settings.BucketName,
```

Comment:
are we sure it is !=nil here always?

Reply:
It should be non-empty from the `Options.Validate` call from `newClient`; however, this made me realize that this might break for the empty client, and thus when running with Azure as the main storage backend.

Reply:
No, wait. We don't know :thinking:
Sorry about the mess.

Reply:
Added a check at the start of the function